category/category_repository_blackbox_test.go

```diff
@@ -0,0 +1,54 @@
+package category_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/almighty/almighty-core/category"
+	"github.com/almighty/almighty-core/gormsupport/cleaner"
+	"github.com/almighty/almighty-core/gormtestsupport"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type categoryRepoBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	repo  category.Repository
+	clean func()
+	ctx   context.Context
+}
+
+func TestRunCategoryRepoBlackBoxTest(t *testing.T) {
+	suite.Run(t, &categoryRepoBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite("../config.yaml")})
+}
+
+func (test *categoryRepoBlackBoxTest) SetupTest() {
+	test.repo = category.NewRepository(test.DB)
+	test.clean = cleaner.DeleteCreatedEntities(test.DB)
+}
+
+func (test *categoryRepoBlackBoxTest) TearDownTest() {
+	test.clean()
+}
+
+// TestCreateCategory tests that we can create a category
+func (test *categoryRepoBlackBoxTest) TestCreateLoadCategory() {
```

Comment:
I have added a few more test cases including those which you have specified above -> f5c2be3  63f95f0    801ed70